c7n/resources/awslambda.py

```diff
@@ -117,6 +117,41 @@ def get_iam_arns(self, resources):
         return [r['Role'] for r in resources]
 
 
+@AWSLambda.filter_registry.register('url-config')
+class URLConfig(ValueFilter):
+
+    annotation_key = "c7n:UrlConfig"
+    schema = type_schema('url-config', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ('lambda:GetFunctionUrlConfig',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('lambda')
+
+        def _augment(r):
+            try:
+                r[self.annotation_key] = self.manager.retry(
+                    client.get_function_url_config, FunctionName=r['FunctionArn'])
+                r[self.annotation_key].pop('ResponseMetadata')
+            except client.exceptions.ResourceNotFoundException:
+                r[self.annotation_key] = {}
+            except ClientError as e:
+                if e.response['Error']['Code'] == ErrAccessDenied:
+                    self.log.warning(
+                        "Access denied getting lambda:%s",
+                        r['FunctionName'])
+                raise
```

Comment:
Do we want to bail before the `raise` here when we hit the access denied case, and possibly set the annotation key like we do in the not found case?

Reply:
two considerations to cover, one around encapsulation of the error handler and logging, re denied, else with embedded iam resources It does feel better to log verbosely and log at the end of the set.

Reply:
Do we need this? I was looking at resource based policies that I could attach in order to DENY a principal on specific lambda actions/api. Looks like the only action that can be denied via resource based policies is `lambda:InvokeFunctionUrl`
A role can potentially lack IAM permissions on lambda read but looks like this piece of code is trying to solve for RBP related blocks. Reading the [add_permissions](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda/client/add_permission.html) doc, I don't see the capability to pass in a `DENY`

Reply:
Also looks like only [these](https://docs.aws.amazon.com/lambda/latest/dg/access-control-resource-based.html) actions are supported in RBP